public/static/app/perf-bb/PerfBBOutliersCtrl.js

```diff
@@ -395,4 +190,373 @@ mciModule.controller('PerfBBOutliersCtrl', function (
       },
     ],
   };
-});
+}).factory('MuteHandler', function(MuteDataService, confirmDialogFactory, $log, $q) {
+  class MuteHandler {
+    constructor(state, lock) {
+      this.state = state;
+      this.lock = lock;
+      this.confirmMuteAction = confirmDialogFactory(function (outliers) {
+        return `Add mute for ${outliers.length} Outliers?`
+      });
+      this.confirmUnmuteAction = confirmDialogFactory(function (outliers) {
+        return `Remove mute for ${outliers.length} Outliers?`
+      });
+    }
+
+    muteSelected() {
+      return _.any(this.state.selection, {muted: true});
+    }
+
+    unmuteSelected() {
+      return _.any(this.state.selection, {muted: false});
+    }
+
+    muteDisabled() {
+      return this.lock.locked || this.state.selection.length === 0 || !this.unmuteSelected();
+    }
+
+    unmuteDisabled() {
+      return this.lock.locked || this.state.selection.length === 0 || !this.muteSelected();
+    }
+
+    muteOutliers(success, error) {
+      const items = this.state.selection;
+      this.lock.lock();
+      this.confirmMuteAction(items)
+        .then(() => {
+          const promises = _.map(items, function (item) {
+            const mute = MuteHandler.getMuteIdentifier(item);
+            if (item.muted) {
+              return mute;
+            } else {
+              return MuteDataService.addMute(mute);
+            }
+          });
+          return $q.all(promises)
+        })
+        .then(success, error)
+        .finally(() => this.lock.unlock())
+        .catch($log.debug);
+    };
+
+    unmuteOutliers(success, error) {
+      const items = this.state.selection;
+      this.lock.lock();
+      this.confirmUnmuteAction(items)
+        .then(() => {
+          const promises = _.map(items, function (item) {
+            const mute = MuteHandler.getMuteIdentifier(item);
+            if (item.muted) {
+              return MuteDataService.unMute(mute);
+            } else {
+              return mute;
+            }
+          });
+          return $q.all(promises)
+        })
+        .then(success, error)
+        .finally(() => this.lock.unlock())
+        .catch($log.debug);
+    };
+
+    static getMuteIdentifier(mute) {
+      return _.pick(mute, "project", "revision", "task", "test", "thread_level", "variant", "create_time", "end", "last_updated_at", "order", "task_id", "version_id");
+    }
+
+  }
+
+  return MuteHandler;
+}).factory('OutlierState', function(FORMAT, MDBQueryAdaptor, EvgUtil, EvgUiGridUtil, uiGridConstants, OutliersDataService, MuteDataService, $q) {
+  class OutlierState {
+    // Create a new state instance to encapsulate filtering, sorting and generating queries.
+    //
+    // :param str project: The project name, e.g. 'sys-perf'.
```

Comment:
This is probably outside of the scope of this PR (and should probably involve the evergreen team), but we might want to consider standardizing on something like [JSDoc formatting](https://en.wikipedia.org/wiki/JSDoc) for javascript code comments.

Reply:
I'm happy to use what ever format is the best once we know what that is. I just copied the format I found in place (which tbh looks remarkebly like sphinx).